### PR TITLE
Improve domain class test coverage

### PR DESCRIPTION
**Pull request by carlschroedl**
Addresses #12 .
Brings us up to 100% line, branch, etc. coverage on the domain classes.

### CODE REVIEW

File: tests/GenericCollectionTestCase.php [hunk 1]
```diff
@@ -5,35 +5,86 @@
 
 abstract class GenericCollectionTestCase extends TestCase
 {
-    protected function assertCollectionReturnsCopyOfArray(GenericCollection $collection, array $originalValues) : void
+    //array of values used to construct the GenericCollection as specified by the subclass
+    protected $values;
+    //GenericCollection instance as specified by the subclass
+    protected $instance;
+    //Concrete type of GenericCollection as specified by the subclass
+    protected $concreteType;
```
Comment (carlschroedl) [inline tests/GenericCollectionTestCase.php:13]: By using a parameterized type we are able to push most of the tests up into this parent test case.

File: tests/GenericCollectionTestCase.php [hunk 2]
```diff
@@ -5,35 +5,86 @@
...
+    {
+        $variousIllegalArguments = array(
+            array(1),
+            array(1,2),
+            //should fail when passed an array of candidate lists.
+            array($this->values)
+        );
+        foreach ($variousIllegalArguments as $illegalArg) {
+            try {
+                $instance = new $this->concreteType($illegalArg);
```
Comment (carlschroedl) [inline tests/GenericCollectionTestCase.php:62]: ^ example usage of the parameterized type

**Review by andrewshell — APPROVED** [review]
Looks good, although I'll have to merge manually because you're PR seems to include duplicates of a bunch of stuff that was already committed. We can review git stuff this weekend.

**Comment by andrewshell** [comment]
Merged as #17

**Comment by carlschroedl** [comment]
Thanks for the cleanup @andrewshell ! Sorry for the mess. It may be a rebase vs merge workflow thing. Let's discuss this weekend.

**Pull request closed without merging**